graph.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+from __future__ import unicode_literals
+
+
+class Graph(object):
+    def __init__(self):
+        self.graph = {}
+
+    def nodes(self):
+        """Returns a list of all nodes in the graph."""
+        return self.graph.keys()
+
+    def edges(self):
+        """Returns a list of all edges in the graph."""
+        edges = []
+        for node, neighbors in self.graph.iteritems():
+            for neighbor in self.graph[node]:
+                edges.append((node, neighbor))
+        return edges
+
+    def add_node(self, new_node, neighbors=[]):
+        """Adds a new node to the graph"""
+        if new_node not in self.nodes():
```

Comment:
You guys have these initial checks in here all over the place. Which is good.. But I would really like to see Try and Except blocks instead.